dask/dataframe/io/parquet.py

```diff
@@ -1107,51 +1056,62 @@ def read_parquet(path, columns=None, filters=None, categories=None, index=None,
-        if path.fn.endswith('_metadata'):
-            # remove '_metadata' from path
-            urlpath = path.fn[:-len('_metadata')]
-        else:
-            urlpath = path.fn
+    if isinstance(columns, str):
+        df = read_parquet(path, [columns], filters, categories, index,
+                          storage_options, engine, infer_divisions)
+        return df[columns]
 
-        fs, fs_token, paths = get_fs_token_paths(
-            urlpath,
-            mode='rb',
-            storage_options=storage_options
-        )
-        paths = path
+    name = 'read-parquet-' + tokenize(path, columns, filters, categories,
+                                      index, storage_options, engine,
+                                      infer_divisions)
+
+    read = get_engine(engine)['read']
+    fs, fs_token, paths = get_fs_token_paths(
+        path, mode='rb',
+        storage_options=storage_options
+    )
+
+    paths = sorted(paths, key=natural_sort_key)  # numeric rather than glob ordering
+
+    func, meta, statistics, parts = read(fs, fs_token, paths, columns=columns, filters=filters,
+                                         categories=categories, index=index,
+                                         gather_statistics=infer_divisions)
+
+    if statistics:
+        parts, statistics = zip(*[(part, stats)
+                                 for part, stats in zip(parts, statistics)
+                                 if stats['num-rows'] > 0])
+
+        # TODO: apply filters
+
+        out = sorted_columns(statistics)
+
+        if len(out) == 1:
+            divisions = out[0]['divisions']
+            index = out[0]['name']
+        elif len(out) > 1:
+            warnings.warn("Multiple sorted columns found: " + ", ".join(o['name'] for o in out))
+            divisions = [None] * (len(parts) + 1)
+        else:
+            divisions = [None] * (len(parts) + 1)
     else:
-        read = get_engine(engine)['read']
-        fs, fs_token, paths = get_fs_token_paths(
-            path, mode='rb',
-            storage_options=storage_options
-        )
+        divisions = [None] * (len(parts) + 1)
 
-        if isinstance(path, string_types) and len(paths) > 1:
-            # Sort paths naturally if multiple paths resulted from a single
-            # specification (by '*' globbing)
-            paths = sorted(paths, key=natural_sort_key)
+    subgraph = {(name, i): (read_parquet_part, func, fs, meta, part['piece'],
+                            list(meta.columns), index, part['kwargs'])
+                for i, part in enumerate(parts)}
 
-    return read(fs, fs_token, paths, columns=columns, filters=filters,
-                categories=categories, index=index, infer_divisions=infer_divisions)
+    return new_dd_object(subgraph, name, meta, divisions)
+
+
+def read_parquet_part(func, fs, meta, part, columns, index, kwargs):
+    """ Read a part of a parquet dataset """
+    df = func(fs, part, columns, **kwargs)
```

Comment:
Comment here to point to the two possible values of func, which would otherwise be tricky to divine here.
If the file is again opened within the pyarrow function (it certainly would be for fastparquet), then OpenFiles would be better than passing around the filesystem (@jcrist , agree on this point?).

Reply:
What do you mean by the two possible values of `func`?  Which values are you referring to?

Reply:
One for pyarrow, one for fastparquet